package/ios/Sources/RNCandle.swift

```diff
@@ -73,52 +78,134 @@ final class HybridRNCandle: HybridRNCandleSpec {
       }
       try viewModel.$linkedAccount
         .removeDuplicates()
-        .compactMap { $0 }
+        .compactMap(\.?.toLinkedAccount)
         .receive(on: RunLoop.main)
-        .sink { [weak self] linkedAccount in
-          guard let self else { return }
-          onSuccess(self.toLinkedAccount(linkedAccount))
-        }
+        .sink(receiveValue: onSuccess)
         .store(in: &cancellables)
     }
   }
 
   // MARK: - Public
 
   public func unlinkAccount(linkedAccountID: String) throws -> Promise<Void> {
-    Promise.async {
+    .async {
       try await self.viewModel.candleClient.unlinkAccount(linkedAccountID: linkedAccountID)
     }
   }
 
   public func getLinkedAccounts() throws -> Promise<[LinkedAccount]> {
-    Promise.async { [weak self] in
-      guard let self else {
-        throw RNClientError.badInitialization(message: "Self was deinitialized \(#function).")
-      }
+    .async {
       let accounts = try await self.viewModel.candleClient.getLinkedAccounts()
-      return accounts.map(self.toLinkedAccount)
+      return accounts.map(\.toLinkedAccount)
     }
   }
 
-  public func getAssetAccounts() throws -> Promise<String> {
+  public func getAssetAccounts(query: AssetAccountQuery) throws -> Promise<[AssetAccount]> {
     .async {
-      return ""
+      let accounts = try await self.viewModel.candleClient.getAssetAccounts(
+        query: .init(
+          linkedAccountIDs: query.linkedAccountIDs,
+          assetKind: .init(rawValue: query.assetKind?.stringValue ?? "")
+        )
+      )
+      return accounts.map { model in
+        let legalAccountKind = LegalAccountKind(fromString: model.legalAccountKind.rawValue)!
+        switch model.details {
+        case .FiatAccountDetails(let fiatDetails):
+          var ach: ACHDetails?
```

Comment:
#req don't use `var`... this kind of thing can just be an inline `ach: fiatDetails.ach.map { achDetails in ... }`